external/kafka-0-10-sql/src/main/scala/org/apache/spark/sql/kafka010/KafkaOffsetReader.scala

```diff
@@ -77,6 +77,19 @@ private[kafka010] trait KafkaOffsetReader {
       partitionTimestamps: Map[TopicPartition, Long],
       failsOnNoMatchingOffset: Boolean): KafkaSourceOffset
 
+  /**
+   * Resolves the specific offsets based on timestamp per all topic-partitions being subscribed.
+   * The returned offset for each partition is the earliest offset whose timestamp is greater
+   * than or equal to the given timestamp in the corresponding partition. If the matched offset
```

Comment:
Nit: `If the matched offset doesn't exist` is a bit odd construct. Either matched or doesn't exist.

Reply:
Let me refine the words. Basically it queries to Kafka and we are explaining the mechanism, it seems OK to say `If Kafka doesn't return the matched offset`.